core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -17223,6 +17223,17 @@ public List<Group> listGroups(Condition condition) throws UserStoreException {
                 "listGroups operation is not implemented in: " + this.getClass());
     }
 
+    /**
+     * Check whether the given group name contain system reserved domain name such as application or internal.
+     *
+     * @param groupName String group name.
+     * @return True if the group name contain system reserved domain name, false otherwise.
+     */
+    private boolean validateGroupDomain(String groupName) {
```

Comment:
Shall we change the method name as we are checking the invalid domains here? `isInvalidGroupDomain`?

Reply:
Fixed